impeller/playground/backend/gles/playground_impl_gles.cc

```diff
@@ -3,35 +3,111 @@
 // found in the LICENSE file.
 
 #include "impeller/playground/backend/gles/playground_impl_gles.h"
+
+#define GLFW_INCLUDE_NONE
+#import "third_party/glfw/include/GLFW/glfw3.h"
+
+#include "flutter/fml/build_config.h"
+#include "impeller/entity/gles/entity_shaders_gles.h"
+#include "impeller/fixtures/gles/fixtures_shaders_gles.h"
+#include "impeller/playground/imgui/gles/imgui_shaders_gles.h"
 #include "impeller/renderer/backend/gles/context_gles.h"
+#include "impeller/renderer/backend/gles/surface_gles.h"
 
 namespace impeller {
 
-PlaygroundImplGLES::PlaygroundImplGLES() = default;
+void PlaygroundImplGLES::DestroyWindowHandle(WindowHandle handle) {
+  if (!handle) {
+    return;
+  }
+  ::glfwDestroyWindow(reinterpret_cast<GLFWwindow*>(handle));
+}
+
+PlaygroundImplGLES::PlaygroundImplGLES()
+    : handle_(nullptr, &DestroyWindowHandle) {
+  ::glfwDefaultWindowHints();
+
+#if FML_OS_MACOSX
+  // ES Profiles are not supported on Mac.
+  ::glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_API);
+#else   // FML_OS_MACOSX
+  ::glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_ES_API);
+  ::glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
+  ::glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+#endif  // FML_OS_MACOSX
+  ::glfwWindowHint(GLFW_RED_BITS, 8);
+  ::glfwWindowHint(GLFW_GREEN_BITS, 8);
+  ::glfwWindowHint(GLFW_BLUE_BITS, 8);
+  ::glfwWindowHint(GLFW_ALPHA_BITS, 8);
+  ::glfwWindowHint(GLFW_DEPTH_BITS, 0);    // no depth buffer
+  ::glfwWindowHint(GLFW_STENCIL_BITS, 8);  // 8 bit stencil buffer
+  ::glfwWindowHint(GLFW_SAMPLES, 0);       // 4xMSAA (disabled)
```

Comment:
Why disable MSAA in the playground?

Reply:
I haven't verified these work in the playground yet and I was trying to simplify the test case to make the frame debugger connect. I'll back this out.

Reply:
Done.